### PR TITLE
Feat/courses

### PR DESCRIPTION
**Pull request by litegral**
New features:

- Fetch course detail
- Fetch all courses
- Create new course
- Update a course
- Delete a course
- Author middleware
- Upload image

Please try the endpoints on Postman. Thanks.

### CODE REVIEW

File: api/controllers/course_controller.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Failed to upload image"})
+		return
+	}
+
+	extension := file.Filename[len(file.Filename)-4:]
+
+	imageID := uuid.New().String()
+
+	filePath := fmt.Sprintf("uploads/%s%s", imageID, extension)
```
Comment (AlbertChoe) [inline api/controllers/course_controller.go:121]: btw ini kamu uploadnya kemana ya? aku tes kok kayaknya urlnya ke local folder yg upload
url nya bukan yg supabase storage kah? @rchronicler

Comment (litegral) [reply api/controllers/course_controller.go:121]: Eh iya aku lupa mention, iya kak itu upload ke local terus diserve static aja /uploads/, keknya gitu lebih simple daripada pakai Supabase Storage, nambah deps lg juga, or I miss something? @AlbertChoe

Comment (AlbertChoe) [reply api/controllers/course_controller.go:121]: oo, bisa sih gitu. cuman kalau banyak course gitu takut localnya full aja si wkwk. Tapi oke la, nanti kalau emg udh selesai, mau diupgrade baru upgrade lgi

**Comment by AlbertChoe** [comment]
nice nice Goodjob

**Pull request merged**